crates/bevy_pbr/src/environment_map/environment_map.wgsl

```diff
@@ -16,14 +18,46 @@ fn environment_map_light(
     N: vec3<f32>,
     R: vec3<f32>,
     F0: vec3<f32>,
+    world_position: vec3<f32>,
 ) -> EnvironmentMapLight {
+    // Search for a reflection probe that contains the fragment.
+    //
+    // TODO(pcwalton): Interpolate between multiple reflection probes.
+    var cubemap_index: i32 = -1;
+    for (var reflection_probe_index: i32 = 0;
+            reflection_probe_index < light_probes.reflection_probe_count;
+            reflection_probe_index += 1) {
+        let reflection_probe = light_probes.reflection_probes[reflection_probe_index];
+        let probe_space_pos =
+            (reflection_probe.inverse_transform * vec4<f32>(world_position, 1.0)).xyz;
+        if (all(abs(probe_space_pos) <= reflection_probe.half_extents)) {
+            cubemap_index = reflection_probe.cubemap_index;
+            break;
+        }
+    }
+
+    // If we didn't find a reflection probe, use the view environment map if applicable.
+    if (cubemap_index < 0) {
+        cubemap_index = light_probes.view_cubemap_index;
+    }
 
+    var out: EnvironmentMapLight;
     // Split-sum approximation for image based lighting: https://cdn2.unrealengine.com/Resources/files/2013SiggraphPresentationsNotes-26915738.pdf
     // Technically we could use textureNumLevels(environment_map_specular) - 1 here, but we use a uniform
     // because textureNumLevels() does not work on WebGL2
     let radiance_level = perceptual_roughness * f32(bindings::lights.environment_map_smallest_specular_mip_level);
-    let irradiance = textureSample(bindings::environment_map_diffuse, bindings::environment_map_sampler, vec3(N.xy, -N.z)).rgb;
-    let radiance = textureSampleLevel(bindings::environment_map_specular, bindings::environment_map_sampler, vec3(R.xy, -R.z), radiance_level).rgb;
+
+    // We always sample the first cubemap to achieve the required control flow uniformity (because of mip levels).
```

Comment:
It would be better to use gradients and calculate the mip level. This is something that we need to do generally. I think maybe @JMS55 did it for the meshlet effort?

Reply:
I don't quite understand what the mip level issue is here. But for the diffuse map, there's only one mipmap level anyways, so feel free to use textureSampleLevel(mip: 0). For the specular map, you need to sample a specific mip based on roughness.

For the meshlet stuff, the gradients I did calculate manually, and patched pbr_fragment.wgsl to sample material textures using them. I don't _think_ there's any other textures I need to change to use explicit gradients samples as all the screen space textures are a single mip level anyways (except for the specular env map, which is different regardless). The only exception is shadow biasing, once we merge #10188.

Reply:
i don't think anything forces diffuse maps to have no mips.

Reply:
The diffuse map should be a tiny 64x64 map at best. There's no reason to have mips - it's just a small LUT. If a user provides one with mips, we should ignore it.

Reply:
I use level 0.0 now and removed the comment.